src/main/java/org/veupathdb/service/eda/ds/plugin/pass/LineplotPlugin.java

```diff
@@ -95,17 +97,40 @@ protected void writeResults(OutputStream out, Map<String, InputStream> dataStrea
     varMap.put("facetVariable1", getVariableSpecFromList(spec.getFacetVariable(), 0));
     varMap.put("facetVariable2", getVariableSpecFromList(spec.getFacetVariable(), 1));
     String showMissingness = spec.getShowMissingness() != null ? spec.getShowMissingness().getValue() : "FALSE";
+    String errorBars = spec.getErrorBars() != null ? spec.getErrorBars().getValue() : "FALSE";
     String valueSpec = spec.getValueSpec().getValue();
+    String xVarType = getVariableType(spec.getXAxisVariable());
+    String numeratorValues = spec.getYAxisNumeratorValues() != null ? listToRVector(spec.getYAxisNumeratorValues()) : "NULL";
+    String denominatorValues = spec.getYAxisDenominatorValues() != null ? listToRVector(spec.getYAxisDenominatorValues()) : "NULL";
     
     useRConnectionWithRemoteFiles(dataStreams, connection -> {
       connection.voidEval(getVoidEvalFreadCommand(DEFAULT_SINGLE_STREAM_NAME, 
           spec.getXAxisVariable(),
           spec.getYAxisVariable(),
           spec.getOverlayVariable(),
           getVariableSpecFromList(spec.getFacetVariable(), 0),
-          getVariableSpecFromList(spec.getFacetVariable(), 1)));
+          getVariableSpecFromList(spec.getFacetVariable(), 1)));          
       connection.voidEval(getVoidEvalVarMetadataMap(DEFAULT_SINGLE_STREAM_NAME, varMap));
-      String cmd = "plot.data::lineplot(" + DEFAULT_SINGLE_STREAM_NAME + ", map, '" + valueSpec + "'," + showMissingness + ")";
+      String viewportRString = getViewportAsRString(spec.getViewport(), xVarType);
+      connection.voidEval(viewportRString);
+      BinSpec binSpec = spec.getBinSpec();
+      validateBinSpec(binSpec, xVarType);
+      // right now its possible to pass type 'numBins' in BinSpec, but we arent supporting it here
+      String binWidth = "NULL";
+        if (xVarType.equals("NUMBER") || xVarType.equals("INTEGER")) {
+          binWidth = binSpec.getValue() == null ? "NULL" : "as.numeric('" + binSpec.getValue() + "')";
+        } else {
+          binWidth = binSpec.getValue() == null ? "NULL" : "'" + binSpec.getValue().toString() + " " + binSpec.getUnits().toString().toLowerCase() + "'";
+        }
+        connection.voidEval("binWidth <- " + binWidth);
+      String cmd = "plot.data::lineplot(" + DEFAULT_SINGLE_STREAM_NAME + 
+                                        ", map, binWidth, " + 
+                                        singleQuote(valueSpec) + 
+                                        ", " + errorBars + 
+                                        ", viewport," + 
+                                        ", " + numeratorValues +
```

Comment:
just checking - right now after `viewport` comes `evilMode` and `collectionVariablePlotRef` (looking at the plot.data PR). Are `numeratorValues` and `denominatorValues` for a later version?